src/plugins/embeddable/public/lib/containers/container.ts

```diff
@@ -39,19 +46,29 @@ export abstract class Container<
     [key: string]: IEmbeddable<any, any> | ErrorEmbeddable;
   } = {};
 
-  private subscription: Subscription;
+  private subscription: Subscription | undefined;
 
   constructor(
     input: TContainerInput,
     output: TContainerOutput,
     protected readonly getFactory: EmbeddableStart['getEmbeddableFactory'],
-    parent?: Container
+    parent?: IContainer,
+    initializeSettings?: EmbeddableContainerInitializeSettings
```

Comment:
non-blocker: one idea I had here is to make this a more generic `options` to give ourselves a backdoor for adding more flags in future (should we every have a need to do so). Then `initializeSettings` is perhaps a setting in options.

Reply:
Good idea, it's an easy change now that might make things easier in the future. Since both sub-keys under the `initializeSettings` have the word `initialize` in them, I've made that type just a generic `ContainerSettings` interface!